src/data/polynomial/coeff.lean

```diff
@@ -105,6 +105,18 @@ begin
   { exact λ h1, (h1 (nat.mem_antidiagonal.2 rfl)).elim }
 end
 
+lemma coeff_mul_X_pow' (p : polynomial R) (n d : ℕ) :
+  (p * X ^ n).coeff d = ite (n ≤ d) (p.coeff (d - n)) 0 :=
+begin
+  by_cases h : n ≤ d,
```

Comment:
Just a stylistic choice, but the `split_ifs` tactic is nice here, and removes the need to `rw` by `if_pos` and `if_neg` later.

Reply:
ooh, I didn't know about that tactic. Let me take a look.